s3fs/core.py

```diff
@@ -18,10 +17,14 @@
 
 logging.getLogger('boto3').setLevel(logging.WARNING)
 logging.getLogger('botocore').setLevel(logging.WARNING)
-S3_RETRYABLE_ERRORS = (
-    socket.timeout,
-    trans.ReadTimeoutError, trans.IncompleteReadError
-)
+
+try:
+    from boto3.s3.transfer import S3_RETRYABLE_ERRORS
+except ImportError:
+    S3_RETRYABLE_ERRORS = (
+        socket.timeout,
```

Comment:
So, merge... and watch this space.
I should have time next week to look into it, if there are no reports between now and then.